File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        unordered_map<string, int> occ;
        vector<string> ans;
        int n = s.size();
        for(int i=0;i<=n-10;++i){
            string subString = s.substr(i, 10);
            ++occ[subString];
        }
        for(auto& x: occ){
            if(x.second > 1){
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.cs
public class Solution {
    public IList<string> FindRepeatedDnaSequences(string s) {
        Dictionary<string, int> occ = new Dictionary<string, int>();
        IList<string> ans = new List<string>();
        int n = s.Length;
        for(int i=0;i<=n-10;i++)
        {
            string subString = s.Substring(i, 10);
            if(occ.ContainsKey(subString)){
                occ[subString]++;
            }
            else
            {
                occ.Add(subString, 1);
            }
        }
        foreach(var x in occ)
        {
            if(x.Value > 1)
            {
                ans.Add(x.Key);
            }
        }
        return ans;
    }
}